> When the server is running and I give the command 'Ctrl + C', the following message is printed in duplicate:
> 
> ```
> info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.479Z"}
> info: Closing http server. {"timestamp":"2019-11-11T13:25:04.482Z"}
> info: Http server closed. {"timestamp":"2019-11-11T13:25:04.483Z"}
> info: SIGINT signal received. {"timestamp":"2019-11-11T13:25:04.483Z"}
> info: Closing http server. {"timestamp":"2019-11-11T13:25:04.484Z"}
> info: Http server closed. {"timestamp":"2019-11-11T13:25:04.484Z"}
> ```
> 
> Also, the error message below is printed out after closing the server using 'Ctrl+C':
> 
> ```
> npm ERR! code ELIFECYCLE
> npm ERR! errno 130
> npm ERR! express-barefoot-nomad@1.0.0 watch: `npm run winston && nodemon --exec babel-node -r ./node_modules/dotenv/config ./src/index.js`
> npm ERR! Exit status 130
> npm ERR! 
> npm ERR! Failed at the express-barefoot-nomad@1.0.0 watch script.
> npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /Users/raymond/.npm/_logs/2019-11-11T13_25_04_511Z-debug.log
> ```
> 
> However, this second error seems to be present in other branches as well and so may not be linked to this PR directly.

I think the error is coming from the Winston logger script  I haven't been able to fix it

--- a/src/app.js
+++ b/src/app.js
@@ -1,0 +1,45 @@
+import express from 'express';
+import { urlencoded, json } from 'body-parser';
+import cors from 'cors';
+import router from './routes/index';
+import logger from './utils/winston';
+import Responses from './utils/response';
+import config from './config';
+
+const isDevelopment = config.env;
+
+const app = express();
+
+app.use(cors());
+
+app.use(urlencoded({ extended: false }));
+app.use(json());
+
+// used in testing heroku
+app.get('/', (req, res) => Responses.handleSuccess(200, 'Welcome to Barefoot Nomad', res));
+
+app.use(router);
+app.use((req, res) => Responses.handleError(404, 'Not found', res));
+
+// development error handler middleware
+app.use((err, req, res, next) => {
+  if (isDevelopment !== 'development') {
+    return next(err);
+  }
+  logger.error(`${err.status || 500} - ${err.message} - ${req.originalUrl} - ${req.method} - ${
+    req.ip
+  } - Stack: ${err.stack}`);
+
+  return Responses.handleError(err.statusCode, `${err.message} - check logs for more info`, res);
+});
+
+// Production and testing error handler middleware
+app.use((err, req, res) => {
+  logger.error(`${err.status || 500} - ${err.message} - ${req.originalUrl} - ${req.method} - ${
+    req.ip
+  } - Stack: ${err.stack}`);
+
+  return Responses.handleError(err.statusCode, err.message, res);
+});
+
+export default app;

--- a/src/index.js
+++ b/src/index.js
@@ -1,60 +1,17 @@
-/* eslint-disable no-unused-vars */
-/* eslint-disable no-console */
-import express from 'express';
-import { json, urlencoded } from 'body-parser';
-import cors from 'cors';
-import router from './routes/index';
+import app from './app';
 import logger from './utils/winston';
-import Responses from './utils/response';
-import ErrorHandler from './utils/error';
 import config from './config';
 
-const isDevelopment = config.env;
-
-
-const isProduction = config.env === 'production';
-
-
-const app = express();
-
-app.use(cors());
-
-app.use(urlencoded({ extended: false }));
-app.use(json());
-
-// used in testing heroku
-app.get('/', (req, res) => Responses.handleSuccess(200, 'Welcome to Barefoot Nomad', res));
-
-// to throw an error to the logger
-app.get('/err', (req, res) => {
-  throw new ErrorHandler('Error here');
+app.set('port', config.PORT || 3000);
+const server = app.listen(app.get('port'), () => {
+  logger.info(`Express running → PORT ${server.address().port}`);
 });
 
-app.use(router);
-app.use((req, res, next) => Responses.handleError(404, 'Not found', res));
-
-// development error handler middleware
-app.use((err, req, res, next) => {
-  if (isDevelopment !== 'development') {
-    return next(err);
-  }
-  logger.error(`${err.status || 500} - ${err.message} - ${req.originalUrl} - ${req.method} - ${
-    req.ip
-  } - Stack: ${err.stack}`);
-
-  return Responses.handleError(err.statusCode, `${err.message} - check logs for more info`, res);
-});
-
-// Production and testing error handler middleware
-app.use((err, req, res, next) => {
-  logger.error(`${err.status || 500} - ${err.message} - ${req.originalUrl} - ${req.method} - ${
-    req.ip
-  } - Stack: ${err.stack}`);
-
-  return Responses.handleError(err.statusCode, err.message, res);
-});
-
-// finally, let's start our server...
-const server = app.listen(config.PORT || 3000, () => {
-  console.log(`Listening on port ${server.address().port}`);
+// Gracefull shut downs
+process.on('SIGTERM', () => {
+  logger.info('SIGTERM signal received.');
+  logger.info('Closing http server.');
+  server.close(() => {
+    logger.info('Http server closed.');
+  });
 });

